Stats  polish: Posting Activity UI in the Insights page

wilmersondasilva opened the pull request:
#### Proposed Changes

* Update the Posting Activity design
* Fix the position of the days (from vertical to horizontal)

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to the Calypso live link
* Navigate to the Insights tab
* Ensure the current site has data available for the Posting Activity section
* Ensure the Posting Activity section is following the design specs

#### Screenshots

Before:
![image](https://user-images.githubusercontent.com/6406071/210022760-338c0794-2961-403d-96f4-fba582e23556.png)
![image](https://user-images.githubusercontent.com/6406071/210022772-56f5edbe-f19a-4d8e-ae93-de25f66ed592.png)

After:
![image](https://user-images.githubusercontent.com/6406071/210022797-ff39cd14-1eaa-42b2-9ea5-d5e65634abd8.png)
![image](https://user-images.githubusercontent.com/6406071/210022807-373b32dc-7017-4762-8d3e-7d52b6f72764.png)

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #70609

wilmersondasilva committed:
Stats: update the Posting Activity UI in the Insigths page

wilmersondasilva committed:
Stats: adjust design details on the Posting Activity UI in the Insights page

wilmersondasilva committed:
Stats: update the Posting Activity tooltip UI in the Insights page

wilmersondasilva commented:
@jeffgolenski I have one question. Some screens don't fit all the months. How can users see all the months in this scenario? Should we add a scroll bar? In the current version, we have arrow buttons on small screens (see the picture).

![image](https://user-images.githubusercontent.com/6406071/210024070-4e5f8455-2fd3-4c1c-a848-43059a563b21.png)

jsnmoon commented on client/my-sites/stats/stats-insights/index.jsx:
```diff
@@ -55,15 +55,17 @@ const StatsInsights = ( props ) => {
+				</div>
+				{ ! isNewAllTimeViews && (
+					<div className="stats__module--insights-unified">
 						<>
 							<SectionHeader label={ translate( 'All-time views' ) } />
 							<StatsViews />
 						</>
```
```suggestion
						<SectionHeader label={ translate( 'All-time views' ) } />
						<StatsViews />
```

We can remove the nesting fragments.

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/day.jsx:
```diff
@@ -29,18 +31,32 @@ class PostTrendsDay extends PureComponent {
-			},
-			comment: 'How many posts published on a certain date.',
-		} );
+		const date = moment( label );
+		const weekDay = date.format( 'dddd' );
+		const formattedDate = date.format( 'MMMM DD, YYYY' );
```
Could we avoid using moment and use [`toLocaleDateString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) instead?

Our dependency on the moment library incurs a heavy bundle size penalty that we're working on alleviating.

wilmersondasilva replied:
Sure! Just used `moment` because I saw it was being used on other components and thought it would be ok. I just implemented the suggested change.

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -128,35 +127,47 @@ class PostTrends extends Component {
-				<SectionHeader label={ this.props.translate( 'Posting activity' ) } />
-				<Card>
+
+				<div>
```
Could we do without nesting the contents in an extra div?
```suggestion
```

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -128,35 +127,47 @@ class PostTrends extends Component {
+					<div className="post-trends__heading">
+						<h1 className="post-trends__title">{ translate( 'Posting activity' ) }</h1>
+
+						{ /* This will be finished later */ }
+						{ /* <YearNavigation
+							disablePreviousArrow={ false }
+							disableNextArrow={ false }
+							onYearChange={ () => {} }
+						/> */ }
```
```suggestion
```

I know this is a draft PR, but let's avoid adding incomplete code as code comments.

wilmersondasilva replied:
Ok! Thanks for pointing it out.

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -128,35 +127,47 @@ class PostTrends extends Component {
+					</div>
+
+					{ /* 
 					<div className={ leftClass } onClick={ this.scrollLeft } role="button" tabIndex="0">
 						<span className="left-arrow" />
 					</div>
 					<div className={ rightClass } onClick={ this.scrollRight } role="button" tabIndex="0">
 						<span className="right-arrow" />
-					</div>
+					</div> */ }
```
Like above, let's delete instead of commenting out unnecessary code.

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/year-navigation.jsx:
```diff
@@ -0,0 +1,43 @@
+import { Icon, arrowLeft, arrowRight } from '@wordpress/icons';
```
Let's avoid committing code that's not rendered anywhere.

jsnmoon replied:
@wilmersondasilva, we should remove this file from this PR before moving forward.

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -163,55 +197,117 @@
+	font-weight: 400;
+	font-size: $font-body-small;
+	line-height: 18px;
+	color: #000;
```
Let's use the corresponding CSS variable for this color (`--color-text`, perhaps?)

jsnmoon replied:
@wilmersondasilva, this comment needs a follow-up.

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -163,55 +197,117 @@
+.tooltip.popover.post-trends-day__tooltip {
+	&.is-top .popover__arrow {
+		border-top-color: var(--color-neutral-100);
+		box-shadow: 0 1px 2px #0000000d;
```
Could we store the box-shadow color as a variable somewhere?

jsnmoon replied:
@wilmersondasilva, what are your thoughts on this?

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -163,55 +197,117 @@
 
-@include breakpoint-deprecated( ">1040px" ) {
+@include breakpoint-deprecated(">1040px") {
+
```
```suggestion
```

(Nit)

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -99,23 +114,39 @@
+	margin-top: 32px;
+	font-family: $font-label;
+	color: var(--color-neutral-80);
+	// --studio-gray-80
```
```suggestion
```

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -73,6 +87,7 @@
 	}
 
 	&.is-active {
+
```
```suggestion
```

(Nit)

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -25,18 +40,17 @@
 	position: relative;
-	height: 130px;
-	max-width: 660px; // necessary for fluid width mode
+	// height: 130px;
+	// max-width: 660px; // necessary for fluid width mode
```
```suggestion
```

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -128,35 +127,47 @@ class PostTrends extends Component {
-		const leftClass = classNames( 'post-trends__scroll-left', {
-			'is-active': this.state.canScrollLeft,
-		} );
+		// const leftClass = classNames( 'post-trends__scroll-left', {
+		// 	'is-active': this.state.canScrollLeft,
+		// } );
```
```suggestion
```

Ditto unused code.

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -128,35 +127,47 @@ class PostTrends extends Component {
-		const rightClass = classNames( 'post-trends__scroll-right', {
-			'is-active': this.state.canScrollRight,
-		} );
+		// const rightClass = classNames( 'post-trends__scroll-right', {
+		// 	'is-active': this.state.canScrollRight,
+		// } );
```
```suggestion
```

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -8,12 +7,12 @@ import { createRef, Component } from 'react';
 import { getSiteStatsPostStreakData } from 'calypso/state/stats/lists/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import Month from './month';
+// import YearNavigation from './year-navigation';
```
```suggestion
```

Let's remove unused code.

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/index.jsx:
```diff
@@ -1,5 +1,4 @@
-import { Card } from '@automattic/components';
-import classNames from 'classnames';
+// import classNames from 'classnames';
```
```suggestion
```

wilmersondasilva replied:
Done!

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -163,55 +197,117 @@
-@include breakpoint-deprecated( ">1040px" ) {
+@include breakpoint-deprecated(">1040px") {
+
 	.post-trends__scroll-left,
 	.post-trends__scroll-right {
 		display: none;
 	}
```
We should clean up obsolete code, either in this PR or in a follow-up.

wilmersondasilva replied:
I removed it on this PR

jsnmoon reviewed (changes requested):
The overall approach looks solid. It would have been helpful to split the tooltip changes into a separate PR, but I'm open to shipping these changes in a single PR.

I see two blockers:

1. We're committing a significant amount of unused code. Let's omit them from this PR.
2. There needs to be more vertical padding for the heading and the footer.

<img width="1539" alt="image" src="https://user-images.githubusercontent.com/4044428/210025708-0861a10f-08dd-4fe9-8a26-30b5ad87ee5c.png">

wilmersondasilva committed:
Stats: add a scroll bar to the Posting Activity section

wilmersondasilva committed:
Stats: fix bug with padding on the Posting Activity section

wilmersondasilva committed:
Stats: remove unused code from Posting Activity section

wilmersondasilva committed:
Stats: remove unnecessary moment.js usage from the Posting Activity section

wilmersondasilva commented:
Thanks for taking a look at it, @jsnmoon.

> 1. We're committing a significant amount of unused code. Let's omit them from this PR.

Most of the unused code was related to the arrow buttons. I was trying to decide what to implement to navigate through the list of months before removing the code. Since the designers are AFK until next week, I added a scroll bar for now because I saw it's the same approach used in other sections. The other part of the unused code was related to showing data from other years. I started to implement this feature, but then I wondered if it would be the case to include it on a different PR, and I paused its implementation halfway through. I was going to remove all the unused code before changing the PR to `Ready to Review`, and I just did it. 

> 2. There needs to be more vertical padding for the heading and the footer.

I fixed it. Thanks for noticing it.

wilmersondasilva commented:
> @jeffgolenski I have one question. Some screens don't fit all the months. How can users see the hidden months in this scenario? Should we add a scroll bar? In the current version, we have arrow buttons on small screens (see the picture).
> 
> ![image](https://user-images.githubusercontent.com/6406071/210024070-4e5f8455-2fd3-4c1c-a848-43059a563b21.png)

For now, I added a horizontal scrollbar because I saw it is a behavior already used on other components of the Insights page, like the `All-time Insights` section. I think it's good enough for now, and we can change it on a follow-up PR if necessary.

jsnmoon reviewed (changes requested):
There are unaddressed review feedback that requires attention before we can move forward with this PR.

Also, it's not possible to view the entire year even when the container is at its maximum width of ~1224px. This is by design, but I wanted to reiterate how unintuitive the interface can be as a consequence of this:

<img width="1396" alt="image" src="https://user-images.githubusercontent.com/4044428/210119247-f26f4f2f-2f0f-46b1-bc27-b7deb1fb1718.png">

<img width="1407" alt="image" src="https://user-images.githubusercontent.com/4044428/210119311-5b9afd3c-28d8-45d7-9cb3-c3be565dd94a.png">

tyxla commented on client/my-sites/stats/post-trends/day.jsx:
```diff
@@ -29,18 +30,36 @@ class PostTrendsDay extends PureComponent {
-			comment: 'How many posts published on a certain date.',
+		const date = new Date( label );
+		const weekDay = date.toLocaleDateString( undefined, { weekday: 'long' } );
+		const formattedDate = date.toLocaleDateString( undefined, {
+			month: 'long',
+			day: 'numeric',
+			year: 'numeric',
```
Noting that for users whose browser language is different than the Calypso interface language, this might display in a different locale, and can end up being a confusing experience. I believe we must match the Calypso interface language at any given time (cc @Automattic/i18n for tips).

wilmersondasilva replied:
Thanks for bringing this up, @tyxla! I saw that in some files we're using `toLocaleDateString( 'en-US', ... )`. Would this be the correct choice here? The other usage of this method that I found on the repo is passing the `locales` param as `undefined` as I did.

dlind1 replied:
As discussed on Slack, I'd suggest using `i18n.getLocaleSlug` here.

tyxla replied:
What @dlind1 suggests makes sense to me, although I'd be happy to blindly trust @Automattic/i18n on a call about what i18n APIs to use.

wilmersondasilva replied:
@dlind1 thanks again for helping me with this. This implementation worked fine and I just pushed it.

wilmersondasilva commented:
> There are unaddressed review feedback that requires attention before we can move forward with this PR.

@jsnmoon, I don't know why, but three of your comments didn't appear to me before. I'll address them now.

> Also, it's not possible to view the entire year even when the container is at its maximum width of ~1224px. This is by design, but I wanted to reiterate how unintuitive the interface can be as a consequence of this.

I agree with you. I implemented it this way because it was on the design specs and the task description. What should I do in this case? I can decrease the day size, and the spacing between the months to fit everything on 1224px. Should I create a new branch/PR from this with this proposal?

wilmersondasilva committed:
Stats: create variables to holds CSS values used on the Posting Activity section

wilmersondasilva committed:
Stats: remove unused file stats/post-trends/year-navigation from this branch

wilmersondasilva commented:
> Also, it's not possible to view the entire year even when the container is at its maximum width of ~1224px. This is by design, but I wanted to reiterate how unintuitive the interface can be as a consequence of this:

I wrote a P2 about it, so we can discuss it over there: stats-posting-activity-ui-p2

wilmersondasilva committed:
Stats: change the Posting Activity UI to show all the months on large screens

wilmersondasilva committed:
Stats: fix the Posting Activity to use getLocaleSlug on toLocaleDateString usage

jsnmoon commented on client/my-sites/stats/post-trends/style.scss:
```diff
@@ -163,55 +136,106 @@
+}
+
+
+.tooltip.popover.post-trends-day__tooltip {
+	&.is-top .popover__arrow {
+		border-top-color: var(--color-neutral-100);
+		box-shadow: 0 1px 2px $tooltip-box-shadow-color;
+	}
+
+	.popover__inner {
+		background-color: var(--color-neutral-100);
+		box-shadow: 0 1px 2px $tooltip-box-shadow-color;
+		border-radius: 4px;
+		padding: 16px 24px;
```
Do we want to migrate some of these styles into the tooltip component in a follow-up PR?

https://github.com/Automattic/wp-calypso/blob/cffafad574ca980b9fd0483b3101ae326dae1bda/client/components/tooltip/style.scss#L177

wilmersondasilva replied:
Sure! This makes total sense to me! I'll create a follow-up PR with this.

wilmersondasilva replied:
This is addressed on this follow-up PR: https://github.com/Automattic/wp-calypso/pull/71676

jsnmoon reviewed (approved):
This is a significant improvement over what we currently have 👍

keoshi reviewed (approved):
This looks great to me and I love the implementation of the tooltip. It makes this section so much more useful and informative.

The only thing I could find while testing the PR was the lack of a margin on the right side on mobile:
<img width="470" alt="image" src="https://user-images.githubusercontent.com/390760/210555854-ed961417-da97-4391-a7a0-d66c96f3befa.png">

I would consider this more of a nice to fix than a blocker though, hence the approval.

wilmersondasilva commented:
> The only thing I could find while testing the PR was the lack of a margin on the right side on mobile

Thanks for pointing out this, @keoshi! I just fixed it on this follow-up PR: https://github.com/Automattic/wp-calypso/pull/71676

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7708878).
